catkin_ws/src/global_perception/scripts/GlobalFuser.py

```diff
@@ -1,74 +1,167 @@
 #!/usr/bin/env python
 
 import rospy
-from std_msgs.msg import String
 from jsk_recognition_msgs.msg import BoundingBoxArray, BoundingBox
 import message_filters as mf
-# from GlobalBBox import GlobalBBoxBuffer
-from global_perception.msg import GlobalBBoxBuffer
-
+import numpy as np
 from jsk_recognition_msgs.msg import BoundingBoxArray
-from global_perception.msg import ChildGlobalBBoxBuffer
 from BBoxPredictor import BBoxPredictor
 
 NODE_RATE = 20 # Hz
-POSITION_TOLERANCE = 0.15 #m
+POSITION_TOLERANCE = 2. #m
+TIME_THRESHOLD = 150
+TOL = 0.5
 
 class GlobalFuser():
     def __init__(self):
         rospy.init_node('global_perception', anonymous=False)
         self.rate = rospy.Rate(NODE_RATE)
 
-        self.sub_n1 = mf.Subscriber('node1/bounding_boxes', BoundingBoxArray)
-        self.sub_n2 = mf.Subscriber('node2/bounding_boxes', BoundingBoxArray)
-        self.sub_n3 = mf.Subscriber('node3/bounding_boxes', BoundingBoxArray)
-        self.time_synch = mf.TimeSynchronizer([self.sub_n1, self.sub_n2, self.sub_n3], 10)
-        self.time_synch.registerCallback(self.nodeCallback)
+        sub_n1 = mf.Subscriber('node1/bounding_boxes', BoundingBoxArray)
+        sub_n2 = mf.Subscriber('node2/bounding_boxes', BoundingBoxArray)
+
+        sub_n12 = mf.Subscriber('node1/bounding_boxes', BoundingBoxArray)
+        # sub_n12.registerCallback(self.testCallback)
+        # time_synch = mf.TimeSynchronizer([sub_n1, sub_n2], 10)
```

Comment:
How come we don't need `TimeSynchronizer` anymore?

Reply:
This was breaking it because of the way the bag is structured. Basically all the bbox messages are sent out in spurts instead of at a constant rate because it's Carla data. So it wouldn't work if I was time syncing things. Needs to be tested on properly spaced out data. 

Tbh I'm not sure how much we need time sync here anyways. Would make it easier from an N node perspective